# U.S. Medical Insurance Costs

In [41]:
# Goal of this project is to find the cost difference between smoker and non-smokers?
# Average cost per individual in each region
# the region with the highest BMI
import csv
import pandas as pd

In [33]:
# variables for all the information in the csv file
age = []
sex = []
BMI = []
children = []
smoker = []
region = []
charges = []

In [42]:
df = pd.read_csv("insurance.csv")
insurance_data = []
with open('insurance.csv', newline = '') as insurance:
    insurance_info = csv.DictReader(insurance)
    for row in insurance_info:
        age.append(row['age'])
        sex.append(row['sex'])
        BMI.append(row['bmi'])
        children.append(row['children'])
        smoker.append(row['smoker'])
        region.append(row['region'])
        charges.append(row['charges'])
  
  
            
      

In [43]:
 # Average cost per individual
 def average_cost_total(charges):
    total_charges = 0
    for cost in charges:
        cost = float(cost)
        total_charges += cost
    average_cost = total_charges/len(charges)
    return int(average_cost)
print("The average cost per individual is " + str(average_cost_total(charges)))  

The average cost per individual is 13270


The average cost per individual is 13270.

In [44]:

average_charges_by_region = df.groupby("region")["charges"].mean()
print(average_charges_by_region)


region
northeast    13406.384516
northwest    12417.575374
southeast    14735.411438
southwest    12346.937377
Name: charges, dtype: float64


The Average cost per region is:
Northeast = 13406.38
Northwest = 12417.58
Southeast = 14735.41
Southwest = 12346.94
The Southeast has the highest average insurance cost.

In [45]:

average_BMI_by_region = df.groupby("region")["bmi"].mean()
print(average_BMI_by_region)

region
northeast    29.173503
northwest    29.199785
southeast    33.355989
southwest    30.596615
Name: bmi, dtype: float64


When looking at the average BMI per region we notice that the Southeast has the highest BMI and that results in the higher cost per average. 

In [46]:
average_charges_by_smoker = df.groupby("smoker")["charges"].mean()
print(average_charges_by_smoker)

smoker
no      8434.268298
yes    32050.231832
Name: charges, dtype: float64


When looking at the cost difference of insurance of smoker vs non smoker the smoker has a much higher cost of insurance. The difference of smoker and non smoker is 23,615.96. That is a huge difference in price of insurance.

In [40]:
smoker_count_by_region = df[df['smoker'] == 'yes'].groupby("region").size().reset_index(name='counts')
print(smoker_count_by_region)

      region  counts
0  northeast      67
1  northwest      58
2  southeast      91
3  southwest      58


There are more smokers in the Southeast region which will increase the price of insurance for the southeast region as well. 